nomad/structs/config/vault.go

```diff
@@ -99,6 +99,58 @@ func (a *VaultConfig) AllowsUnauthenticated() bool {
 	return a.AllowUnauthenticated != nil && *a.AllowUnauthenticated
 }
 
+// IsEqual compares two Vault configurations and returns a boolean indicating
+// if they are equal.
+func (a *VaultConfig) IsEqual(b *VaultConfig) bool {
```

Comment:
Use this to make Vault reloading more precise: https://github.com/hashicorp/nomad/blob/master/nomad/server.go#L675